src/NuGet.Jobs.GitHubIndexer/RepoUtils.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+using NuGet.Packaging;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class RepoUtils
+    {
+        public class GitFileInfo
+        {
+            public GitFileInfo(string path, long blobSize)
+            {
+                Path = path;
+                BlobSize = blobSize;
+            }
+
+            public string Path { get; }
+            public long BlobSize { get; }
+        }
+
+        private readonly ILogger<RepoUtils> _logger;
+
+        public RepoUtils(ILogger<RepoUtils> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Returns all the files at a specific commit by listing the file tree
+        /// </summary>
+        /// <param name="tree">The file tree</param>
+        /// <param name="currentPath">The initial file path. (Default is "")</param>
+        /// <param name="repo">The repository containing the file tree</param>
+        /// <returns>List of nodes that are in the file tree.</returns>
+        public IReadOnlyList<GitFileInfo> ListTree(LibGit2Sharp.Tree tree, string currentPath, LibGit2Sharp.Repository repo)
+        {
+            var files = new List<GitFileInfo>();
+            foreach (var node in tree)
+            {
+                var nodePath = string.IsNullOrWhiteSpace(currentPath) ?
+                    node.Path :
+                    currentPath + Path.DirectorySeparatorChar + node.Path;
+
+                if (node.TargetType == LibGit2Sharp.TreeEntryTargetType.Tree)
+                {
+                    files.AddRange(ListTree((LibGit2Sharp.Tree)repo.Lookup(node.Target.Id), nodePath, repo));
+                    continue;
+                }
+
+                // Only handle files
+                if (node.TargetType != LibGit2Sharp.TreeEntryTargetType.Blob)
+                {
+                    continue;
+                }
+
+                var blobSize = (node.Target as LibGit2Sharp.Blob).Size;
```

Comment:
Prefer casting over `as`. If the `.Target` is ever not a `Blob`, this will throw a null reference exception and hide the real issue.

```suggestion
                var blobSize = ((LibGit2Sharp.Blob)node.Target).Size;
```